### PR TITLE
TRAN-5593: Update PATH arrivals translator for new track-specific arrivals

### PR DESCRIPTION
**Pull request by JorgeBillini**
## Changelog
Updates PATH arrivals translator with new private API endpoint with track specific targeting. 
Adds new mappings for Arrivals endpoint based on documentation provided by PATH

### CODE REVIEW

File: gtfs_realtime_translators/translators/path_rail.py [hunk 1]
```diff
@@ -109,33 +149,67 @@ def __call__(self, data):
...
+    @classmethod
+    def __route_lookup(cls,service_id, station_shortkey, track_id, destination):
+        service_mapping = cls.SERVICE_LOOKUP.get(str(service_id))
+        station_mapping = cls.STOP_ID_LOOKUP.get(station_shortkey)
+        route_id = service_mapping.get('route_id')
+        headsign = service_mapping.get(destination)
+        stop_id = station_mapping.get("tracks").get(track_id)
+        stop_name = station_mapping.get("stop_name")
+        return {
+           'route_id': route_id,
```
Comment (piyushgovani) [inline gtfs_realtime_translators/translators/path_rail.py:165]: Looks like one space lesser than next line? Or it is just git showing that way?

File: gtfs_realtime_translators/translators/path_rail.py [hunk 2]
```diff
@@ -109,33 +149,67 @@ def __call__(self, data):
...
+        for station in stations:
+            station_shortkey = station['abbrv']
+            tracks = station.get('tracks')
+            if station_shortkey == 'systemwide':
+                continue
+            for track in tracks:
+                track_id = track.get('trackId')
+                trains = track.get('trains')
+                for idx,train in enumerate(trains):
+                    if(not train.get('trainId')):
```
Comment (piyushgovani) [inline gtfs_realtime_translators/translators/path_rail.py:193]: better to have:
if not train.get('trainId'):

File: gtfs_realtime_translators/translators/path_rail.py [hunk 3]
```diff
@@ -109,33 +149,67 @@ def __call__(self, data):
...
+                        continue
+                    train_info = train.get('trainId').split('_')
+                    service_id = train.get('service')
+                    destination = train.get('destination')
+                    arrival_time = train.get('depArrTime')
+                    scheduled_arrival_time = cls.__to_unix_time(train_info[0])
+                    arrival_data = cls.__route_lookup(service_id,station_shortkey,track_id, destination)
+                    if(cls.__should_skip_update(arrival_data)):
```
Comment (piyushgovani) [inline gtfs_realtime_translators/translators/path_rail.py:201]: Same as above.
if cls.__should_skip_update(arrival_data):

File: gtfs_realtime_translators/translators/path_rail.py [hunk 4]
```diff
@@ -109,33 +149,67 @@ def __call__(self, data):
...
+        trip_updates = []
+        stations = data['stations']
+        for station in stations:
+            station_shortkey = station['abbrv']
+            tracks = station.get('tracks')
+            if station_shortkey == 'systemwide':
+                continue
+            for track in tracks:
+                track_id = track.get('trackId')
+                trains = track.get('trains')
```
Comment (piyushgovani) [inline gtfs_realtime_translators/translators/path_rail.py:191]: Can have as below:
trains = track.get('trains', {})

Reason is: trains = track.get('trains') gives trains as None (unless, it is initialized while creating trains in track) and that will mostly result in exception at next line while using it as `enumerate(trains)`

**Review by piyushgovani — COMMENTED** [review]
Nice transformation...

File: gtfs_realtime_translators/translators/path_rail.py (hunk 1 above)
Comment (JorgeBillini) [reply gtfs_realtime_translators/translators/path_rail.py:165]: It might just be git! my text editor shows the same amount of space

**Comment by JorgeBillini** [comment]
> Nice transformation...

Thank you! Addressed your feedback, and modified the entity_id field to be unique rather than the array index.

**Review by piyushgovani — APPROVED** [review]

**Review by paulgasbarra — APPROVED** [review]

**Pull request merged**